geowebcache/core/src/main/java/org/geowebcache/util/Logging.java

```diff
@@ -0,0 +1,120 @@
+/**
+ * This program is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Lesser General Public License as published by the Free Software Foundation, either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * <p>You should have received a copy of the GNU Lesser General Public License along with this
+ * program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author Jody Garnett - GeoCat 2022
+ */
+package org.geowebcache.util;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.core.LoggerContext;
+import org.apache.logging.log4j.core.config.Configuration;
+import org.apache.logging.log4j.core.config.DefaultConfiguration;
+import org.apache.logging.log4j.core.config.NullConfiguration;
+import org.apache.logging.log4j.core.config.builder.impl.BuiltConfiguration;
+import org.apache.logging.log4j.core.config.json.JsonConfiguration;
+import org.apache.logging.log4j.core.config.properties.PropertiesConfiguration;
+import org.apache.logging.log4j.core.config.xml.XmlConfiguration;
+import org.apache.logging.log4j.core.config.yaml.YamlConfiguration;
+
+/**
+ * Uses {@link org.geotools.util.logging.Logging} to delegate to provided logging environment.
```

Comment:
I don't understand why this class is needed... all you need is a web application initializer setting the preferred logging framework (you can contorl the order of the initializers execution in web.xml) and would have avoided having to setup logs in a way that's different from GeoTools.
Also, this setup is quite error-prone, when writing a new class people used to GeoTools will likely just use GeoTools own Logging instead.

I'd strongly recommend to back out this change, if not all, at least having to go through gwc Logging rather than Geotools logging.

Reply:
Does it happen too late in GeoServer? Has worked fine for years, happens in LoggingStartupContextListener

Reply:
Andrea the solid reason for the new Logging class is to respect GeoWebCache convention of using the full class name, GeoTools uses just the package name.

Reply:
That's just aesthetics... how the log messages end up showing in the log files should be up to a user-controllable configuration.

Reply:
This class is removed, LoggingContextListener introduced providing geoserver like control with system properties.

Indeed ``org.geowebcache.util.logging.policy`` policy of ``Provided`` avoid any logging configuration; intended for when geowebcache is used by geoserver as a library.